crates/communication/src/server/acceptor.rs

```diff
@@ -27,15 +27,22 @@ pub enum AcceptError {
     ConnectionsErrored(Vec<ConnectionError>),
 }
 
-pub fn acceptor(
+pub(crate) fn acceptor(
+    addresses: Option<impl ToSocketAddrs + Send + Sync + 'static>,
     keep_running: CancellationToken,
     outputs_sender: Sender<outputs::Request>,
+    parameters_sender: Sender<parameters::ClientRequest>,
 ) -> JoinHandle<Result<(), AcceptError>> {
     let next_client_id = AtomicUsize::default();
     spawn(async move {
+        let Some(addresses) = addresses else {
+            // requested to disable Communication socket
+            return Ok(());
+        };
```

Comment:
Why do you durchschleifen this addresses value all the way through here?
Couldn't this branch live in the runtime? Don't need to spawn an acceptor task if it exits immediately anyways.

I always find it weird to pass Options around that if None cause the functions to return immediately unless passing the Option is required by some Trait bound.

Reply:
Moved into Runtime